src/scenes/RSVP/index.jsx

```diff
@@ -0,0 +1,128 @@
+import React from 'react';
+import './style.scss';
+
+import {
+  getRegistration,
+  getRoles,
+  register,
+  rsvp,
+} from 'api';
+
+import { Form, Formik } from 'formik';
+
+import Loading from 'components/Loading';
+import Message from 'components/Message';
+import SelectField from 'components/SelectField';
+import SubmitButton from 'components/SubmitButton';
+
+export default class MentorRegistration extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isLoading: true,
+      isEditing: false,
+      isSubmitted: false,
+
+      registration: {},
+    };
+  }
+
+  componentDidMount() {
+    getRoles().then(roles => {
+      if (roles.includes('Applicant')) {
+        this.setState({ isEditing: true });
+        return getRegistration('attendee');
+      }
+      return {};
+    }).then(registration => {
+      this.setState({ registration });
+    }).finally(() => {
+      this.setState({ isLoading: false });
+    });
+  }
+
+  submit = registration => {
+    const { isEditing } = this.state;
+    this.setState({ registration, isLoading: true });
+
+    register(isEditing, 'attendee', registration).then(rsvp()).then(() => {
+      this.setState({ isLoading: false, isSubmitted: true });
+    });
+  }
+
+  page = ({ values }) => {
+    const isValid = values.shirtSize && values.hasDisbility !== '';
```

Comment:
Typo
```suggestion
    const isValid = values.shirtSize && values.hasDisability !== '';
```

Also, it lets me leave the hasDisability field blank initially because it isn't part of `values` (`values.hasDisability` is `undefined`, so `isValid` can be true)
But if I modify the field and then clear, then the submit button is disabled (because `values.hasDisability` is then `""`)
Is this intended or is there any other reason why you didn't just do the following?
```suggestion
    const isValid = values.shirtSize && values.hasDisability;
```

Reply:
Oh, `hasDisability` can be `false` so you can't do that. Maybe this instead:
```suggestion
    const isValid = values.shirtSize && typeof values.hasDisability === 'boolean';
```

Reply:
I'm going to keep this how it is for now just to stay consistent. In the future, it would be nice to have actual form validation (possibly with yup)